app/controllers/cars_controller.rb

```diff
@@ -0,0 +1,35 @@
+class CarsController < ApplicationController
+  before_action :find_current_user, except: %i[show destroy]
+  before_action :find_car, only: %i[show destroy]
+
+  def index
+    render_response(@user.cars)
+  end
+
+  def show
+    render_response(@car)
+  end
+
+  def create
+    @user.cars.create(car_params)
+    render_success_response(data = nil, status = :created)
+  end
+
+  def destroy
+    render_success_response(data = nil, status = :no_content) if @car.destroy
+  end
+
+  private
+
+  def car_params
+    params.require(:car).permit(:manufacturer, :model, :year, :picture, :color, :license_plat_number)
+  end
+
+  def find_current_user
+    @user = Users::CurrentUserService.perform(request.headers['token'])
```

Comment:
replace service call with service content and delete `Users::CurrentUserService`
rename `find_current_user` to `current_user` and move it to `ApplicationController`